Practice/pig_latin.py

```diff
@@ -2,81 +2,75 @@
-        pig_latin_word = ''.join(letter_list[3:-1]) + ''.join(letter_list[0:3]) + 'ay' + letter_list[-1]
-    else:
-        pig_latin_word = ''.join(letter_list[3:]) + ''.join(letter_list[0:3]) + 'ay'
-    word_list[0] = pig_latin_word
-
-i = 1  # loop through the words until end of the sentence
-while i < len(word_list):
-    letter_list = list(str(word_list[i]))
-    if letter_list[0] in FIRST_LETTER_VOWELS:
-        if letter_list[-1] in PUNCTUATION:
-            pig_latin_word = ''.join(letter_list[:-1]) + 'ay' + letter_list[-1]
-        else:
-            pig_latin_word = ''.join(letter_list) + 'ay'
-        word_list[i] = pig_latin_word
-
-    elif letter_list[1] in OTHER_LETTER_VOWELS:
-        if str(letter_list[0]).isupper():
-            letter_list[0] = str(letter_list[0]).lower()
-            letter_list[1] = str(letter_list[1]).upper()
-        if letter_list[-1] in PUNCTUATION:
-            pig_latin_word = ''.join(letter_list[1:-1]) + letter_list[0] + 'ay' + letter_list[-1]
-        else:
-            pig_latin_word = ''.join(letter_list[1:]) + letter_list[0] + 'ay'
-        word_list[i] = pig_latin_word
-    elif letter_list[2] in OTHER_LETTER_VOWELS:
-        if str(letter_list[0]).isupper():
-            letter_list[0] = str(letter_list[0]).lower()
-            letter_list[2] = str(letter_list[2]).upper()
-        if letter_list[-1] in PUNCTUATION:
-            pig_latin_word = ''.join(letter_list[2:-1]) + ''.join(letter_list[0:2]) + 'ay' + letter_list[-1]
+        end_of_word = word[index:-1]
+        word = end_of_word + letters_to_move.lower() + 'ay'
+    return word
+
+
+def find_index_of_first_vowel(word):
+    """Returns the index of the first vowel in the word"""
+    if word[0] in FIRST_LETTER_VOWELS:
+        return 0
+    elif word[1] in FIRST_LETTER_VOWELS:
+        return 1
+    elif word[2] in FIRST_LETTER_VOWELS:
+        return 2
     else:
-        if str(letter_list[0]).isupper():
-            letter_list[0] = str(letter_list[0]).lower()
-            letter_list[3] = str(letter_list[3]).upper()
-        if letter_list[-1] in PUNCTUATION:
-            pig_latin_word = ''.join(letter_list[3:-1]) + ''.join(letter_list[0:3]) + 'ay' + letter_list[-1]
-        else:
-            pig_latin_word = ''.join(letter_list[3:]) + ''.join(letter_list[0:3]) + 'ay'
-        word_list[i] = pig_latin_word
-    i += 1
-pig_latin_sentence = ' '.join(word_list)
-print(sentence + ' in Pig Latin is ' + pig_latin_sentence)
+        return 3
+
+
+def convert_to_pig_latin(word_list):
+    """takes a list of words, converts the words to pig latin, then rejoins them into a sentence"""
+    for index in range(len(word_list)):
+        first_vowel_index = find_index_of_first_vowel(word_list[index])
+        word_list[index] = move_letters(word_list[index], first_vowel_index)
+
+    return word_list
+
+
+def convert_list_to_string(pig_latin_list):
+    sentence = ' '.join(pig_latin_list)
+    return sentence
+
+def input_sentence():
```

Comment:
PEP-8 says 2 blank lines between top level functions. Listen to any warnings the linter might give you and fix them all.